tests/unit/metrics/aupro/aupro_reference.py

```diff
@@ -56,29 +62,31 @@ def compute_overlap(self, threshold):
         return 1.0 - self.index / len(self.anomaly_scores)
 
 
-def trapezoid(x, y, x_max=None):
-    """
-    This function calculates the definit integral of a curve given by x- and corresponding y-values.
+def trapezoid(x: list[float], y: list[float], x_max: float | None = None) -> float:
+    """This function calculates the definit integral of a curve given by x- and corresponding y-values.
+
     In contrast to, e.g., 'numpy.trapz()', this function allows to define an upper bound to the integration range by
     setting a value x_max.
     Points that do not have a finite x or y value will be ignored with a warning.
+
     Args:
         x:     Samples from the domain of the function to integrate need to be sorted in ascending order. May contain
                the same value multiple times. In that case, the order of the corresponding y values will affect the
                integration with the trapezoidal rule.
         y:     Values of the function corresponding to x values.
         x_max: Upper limit of the integration. The y value at max_x will be determined by interpolating between its
                neighbors. Must not lie outside of the range of x.
+
     Returns:
         Area under the curve.
     """
-
     x = np.array(x)
     y = np.array(y)
     finite_mask = np.logical_and(np.isfinite(x), np.isfinite(y))
     if not finite_mask.all():
-        print(
-            "WARNING: Not all x and y values passed to trapezoid are finite. Will continue with only the finite values."
+        logger.warning(
```

Comment:
does pre-commit like this? Thought it would want a message variable

Reply:
I didn't get any issue with pre-commit for this one. I think this is required only for Exceptions.